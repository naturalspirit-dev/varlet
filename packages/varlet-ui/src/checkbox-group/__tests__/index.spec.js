const CheckboxGroup = require('../../../cjs/checkbox-group').default
const { render } = require('@testing-library/vue')

test('test checkboxGroup', async () => {
  const wrapper = render(CheckboxGroup)
  console.log(wrapper)
})
