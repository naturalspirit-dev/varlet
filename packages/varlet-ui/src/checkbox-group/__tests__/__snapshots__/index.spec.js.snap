// Vitest Snapshot v1

exports[`test checkbox group layout direction 1`] = `
"<div class=\\"var-checkbox-group__wrap\\">
  <div class=\\"var-checkbox-group var-checkbox-group--vertical\\">
    <div class=\\"var-checkbox__wrap\\">
      <div class=\\"var-checkbox\\">
        <div class=\\"var-checkbox__action var-checkbox--unchecked\\"><i class=\\"var-icon var-icon--set var-icon-checkbox-blank-outline var-checkbox__icon\\" style=\\"transition: transform 0ms;\\" var-checkbox-cover=\\"\\"></i></div>
        <div class=\\"var-checkbox__text\\"></div>
      </div>
      <transition-stub>
        <!--v-if-->
      </transition-stub>
    </div>
    <div class=\\"var-checkbox__wrap\\">
      <div class=\\"var-checkbox\\">
        <div class=\\"var-checkbox__action var-checkbox--unchecked\\"><i class=\\"var-icon var-icon--set var-icon-checkbox-blank-outline var-checkbox__icon\\" style=\\"transition: transform 0ms;\\" var-checkbox-cover=\\"\\"></i></div>
        <div class=\\"var-checkbox__text\\"></div>
      </div>
      <transition-stub>
        <!--v-if-->
      </transition-stub>
    </div>
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`test checkbox group validation 1`] = `
"<div class=\\"var-checkbox-group__wrap\\">
  <div class=\\"var-checkbox-group var-checkbox-group--horizontal\\">
    <div class=\\"var-checkbox__wrap\\">
      <div class=\\"var-checkbox\\">
        <div class=\\"var-checkbox__action var-checkbox--unchecked var-checkbox--error\\"><i class=\\"var-icon var-icon--set var-icon-checkbox-blank-outline var-checkbox__icon\\" style=\\"transition: transform 0ms;\\" var-checkbox-cover=\\"\\"></i></div>
        <div class=\\"var-checkbox__text var-checkbox--error\\"></div>
      </div>
      <transition-stub>
        <!--v-if-->
      </transition-stub>
    </div>
    <div class=\\"var-checkbox__wrap\\">
      <div class=\\"var-checkbox\\">
        <div class=\\"var-checkbox__action var-checkbox--unchecked var-checkbox--error\\"><i class=\\"var-icon var-icon--set var-icon-checkbox-blank-outline var-checkbox__icon\\" style=\\"transition: transform 0ms;\\" var-checkbox-cover=\\"\\"></i></div>
        <div class=\\"var-checkbox__text var-checkbox--error\\"></div>
      </div>
      <transition-stub>
        <!--v-if-->
      </transition-stub>
    </div>
  </div>
  <transition-stub>
    <div class=\\"var-form-details\\">
      <div class=\\"var-form-details__message\\">至少选一个</div>
      <div class=\\"var-form-details__length\\"></div>
    </div>
  </transition-stub>
</div>"
`;

exports[`test checkbox group validation 2`] = `
"<div class=\\"var-checkbox-group__wrap\\">
  <div class=\\"var-checkbox-group var-checkbox-group--horizontal\\">
    <div class=\\"var-checkbox__wrap\\">
      <div class=\\"var-checkbox\\">
        <div class=\\"var-checkbox__action var-checkbox--checked\\"><i class=\\"var-icon var-icon--set var-icon-checkbox-marked var-checkbox__icon var-checkbox--with-animation\\" style=\\"transition: transform 0ms;\\" var-checkbox-cover=\\"\\"></i></div>
        <div class=\\"var-checkbox__text\\"></div>
      </div>
      <transition-stub>
        <!--v-if-->
      </transition-stub>
    </div>
    <div class=\\"var-checkbox__wrap\\">
      <div class=\\"var-checkbox\\">
        <div class=\\"var-checkbox__action var-checkbox--unchecked\\"><i class=\\"var-icon var-icon--set var-icon-checkbox-blank-outline var-checkbox__icon\\" style=\\"transition: transform 0ms;\\" var-checkbox-cover=\\"\\"></i></div>
        <div class=\\"var-checkbox__text\\"></div>
      </div>
      <transition-stub>
        <!--v-if-->
      </transition-stub>
    </div>
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`test checkbox validation 1`] = `
"<div class=\\"var-checkbox__wrap\\">
  <div class=\\"var-checkbox\\">
    <div class=\\"var-checkbox__action var-checkbox--unchecked var-checkbox--error\\"><i class=\\"var-icon var-icon--set var-icon-checkbox-blank-outline var-checkbox__icon\\" style=\\"transition: transform 0ms;\\" var-checkbox-cover=\\"\\"></i></div>
    <div class=\\"var-checkbox__text var-checkbox--error\\"></div>
  </div>
  <transition-stub>
    <div class=\\"var-form-details\\">
      <div class=\\"var-form-details__message\\">您必须勾选</div>
      <div class=\\"var-form-details__length\\"></div>
    </div>
  </transition-stub>
</div>"
`;

exports[`test checkbox validation 2`] = `
"<div class=\\"var-checkbox__wrap\\">
  <div class=\\"var-checkbox\\">
    <div class=\\"var-checkbox__action var-checkbox--checked\\"><i class=\\"var-icon var-icon--set var-icon-checkbox-marked var-checkbox__icon var-checkbox--with-animation\\" style=\\"transition: transform 0ms;\\" var-checkbox-cover=\\"\\"></i></div>
    <div class=\\"var-checkbox__text\\"></div>
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
</div>"
`;
